lib/active_record_host_pool/connection_adapter_mixin.rb

```diff
@@ -118,3 +181,4 @@ def establish_connection(owner, spec)
 # rubocop:enable Lint/DuplicateMethods
 
 ActiveRecord::ConnectionAdapters::Mysql2Adapter.include(ActiveRecordHostPool::DatabaseSwitch)
+ActiveRecord::ConnectionAdapters::Mysql2Adapter.include(ActiveRecordHostPool::ThreadSafe)
```

Comment:
I made `ThreadSafe` a separate module in case this functionality should be considered opt-in. 

If this is made opt-in it may make sense to separate test thread-safety separate from non-thread-safety (to run separately when we modify the Mysql2Adapter with thread-safety from running the original non-thread-safe code), but I did not tackle the in this PR since it may not be necessary.